plaso/parsers/google_logging.py

```diff
@@ -0,0 +1,262 @@
+
+  Attributes:
+    file_name (str): the name of the source file that logged the message.
+    line_number (int): the line number in the source file where the logging
+      statement is.
+    message (str): the log message.
+    priority (str): the priority of the message - I, W, E or F.
+    thread_id (int): the ID of the thread that recorded the message.
+  """
+
+  DATA_TYPE = 'googlelog:log'
+
+  def __init__(self, data_type=DATA_TYPE):
+    """Initializes the event object.
+
+    Args:
+      data_type (Optional[str]): event data type indicator.
+    """
+    super(GoogleLogEventData, self).__init__(data_type=data_type)
+    self.file_name = None
+    self.line_number = None
+    self.message = None
+    self.priority = None
+    self.thread_id = None
+
+
+class GoogleLogParser(text_parser.PyparsingMultiLineTextParser):
+  """A parser for Google log formatted files.
+
+  See https://google.io/docs/python/guides/logging
+  """
+  NAME = 'googlelog'
+
+  DESCRIPTION = 'Parser for handling Google log formatted files.'
+
+  # PyParsing components used to construct grammars for parsing lines.
+  _PYPARSING_COMPONENTS = {
+      'priority':
+          (pyparsing.oneOf(['I', 'W', 'E', 'F']).setResultsName('priority')),
+      'year': text_parser.PyparsingConstants.FOUR_DIGITS.setResultsName(
+          'year'),
+      'month_number': text_parser.PyparsingConstants.TWO_DIGITS.setResultsName(
+          'month_number'),
+      'day': text_parser.PyparsingConstants.ONE_OR_TWO_DIGITS.setResultsName(
+          'day'),
+      'hour': text_parser.PyparsingConstants.TWO_DIGITS.setResultsName(
+          'hour'),
+      'minute': text_parser.PyparsingConstants.TWO_DIGITS.setResultsName(
+          'minute'),
+      'second': text_parser.PyparsingConstants.TWO_DIGITS.setResultsName(
+          'second'),
+      'microsecond': pyparsing.Word(pyparsing.nums, exact=6).setParseAction(
+          text_parser.PyParseIntCast).setResultsName('microsecond'),
+      'thread_id': pyparsing.Word(pyparsing.nums).setResultsName('thread_id'),
+      'file_name':
+          (pyparsing.Word(pyparsing.printables.replace(':', '')).setResultsName(
+              'file_name')),
+      'line_number':
+          (pyparsing.Word(pyparsing.nums).setResultsName('line_number')),
+      'message':
+          (pyparsing.Regex('.*?(?=($|\n[IWEF][0-9]{4}))', re.DOTALL).
+           setResultsName('message'))}
+
+  _PYPARSING_COMPONENTS['date'] = (
+      _PYPARSING_COMPONENTS['month_number'] +
+      _PYPARSING_COMPONENTS['day'] +
+      _PYPARSING_COMPONENTS['hour'] + pyparsing.Suppress(':') +
+      _PYPARSING_COMPONENTS['minute'] + pyparsing.Suppress(':') +
+      _PYPARSING_COMPONENTS['second'] + pyparsing.Optional(
+          pyparsing.Suppress('.') +
+          _PYPARSING_COMPONENTS['microsecond']))
+
+  # Grammar for individual log event lines.
+  _LOG_LINE = (
+      _PYPARSING_COMPONENTS['priority'] + _PYPARSING_COMPONENTS['date']
+      + _PYPARSING_COMPONENTS['thread_id'] +
```

Comment:
Why the `+` at the start of the line, where everywhere else it is at the end?
Please change to make this consistent.